libs/wingsdk/test/shared-gcp/bucket.inflight.test.ts

```diff
@@ -0,0 +1,341 @@
+  const res = await client.exists(KEY);
+
+  expect(res).toBe(false);
+});
+
+test("tryGet an existing object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const KEY = "test-file-1";
+  const VALUE = "test-content-1";
+
+  const storage = new MockStorage();
+  await storage.bucket(BUCKET_NAME).put(KEY, VALUE);
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryGet(KEY);
+
+  expect(res).toBe(VALUE);
+});
+
+test("tryGet a non-existent object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const NON_EXISTENT_KEY = "test-file-1";
+
+  const storage = new MockStorage();
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryGet(NON_EXISTENT_KEY);
+
+  expect(res).toBe(undefined);
+});
+
+test("tryGetJson an existing object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const KEY = "test-file-1";
+  const VALUE = { cool: "beans" };
+
+  const storage = new MockStorage();
+  await storage.bucket(BUCKET_NAME).put(KEY, JSON.stringify(VALUE));
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryGetJson(KEY);
+
+  expect(res).toEqual(VALUE);
+});
+
+test("tryGetJson a non-existent object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const NON_EXISTENT_KEY = "NON_EXISTENT_KEY";
+
+  const storage = new MockStorage();
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryGetJson(NON_EXISTENT_KEY);
+
+  expect(res).toBe(undefined);
+});
+
+test("tryGetJson an existing non-Json object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const KEY = "test-file-1";
+  const NON_JSON_VALUE = "test-content-1";
+
+  const storage = new MockStorage();
+  await storage.bucket(BUCKET_NAME).put(KEY, NON_JSON_VALUE);
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+
+  await expect(() => client.tryGetJson(KEY)).rejects.toThrowError(
+    `Failed to tryGet JSON object. (key=${KEY})`
+  );
+});
+
+test("tryDelete an existing object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const KEY = "test-file-1";
+  const VALUE = "test-content-1";
+
+  const storage = new MockStorage();
+  await storage.bucket(BUCKET_NAME).put(KEY, VALUE);
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryDelete(KEY);
+
+  expect(res).toBe(true);
+});
+
+test("tryDelete a non-existent object from the bucket", async () => {
+  const BUCKET_NAME = "test-bucket";
+  const NON_EXISTENT_KEY = "NON_EXISTENT_KEY";
+
+  const storage = new MockStorage();
+
+  const client = new BucketClient(BUCKET_NAME, storage as any);
+  const res = await client.tryDelete(NON_EXISTENT_KEY);
+
+  expect(res).toBe(false);
+});
+
+// test("Given a bucket, when giving one of its keys, we should get its signed url", async () => {
```

Comment:
nit: If possible can we avoid including unused code? To save code for later I recommend saving it in a separate branch